### DISCUSSION TITLE
How to integrate Alltalk_tts  into my next.js project?

### DISCUSSION THREAD

**Discussion by arron122** [erew123/alltalk_tts#357] [Q&A] [answered]
Noob here 👋. My objective is to create a virtual Ai-assistant that that can take queries and reply in both text and speech. I already managed to connect Ollama, into my next.js project, and get written responses. Trying to figure out how to integrate Alltalk_tts into my project.
Any directions on how to accomplish this would be much appreciated 👀.

**Comment by erew123** [comment, marked as answer]
Hi @arron122 

Have you looked at the API documentation? https://github.com/erew123/alltalk_tts/wiki#-api-documentation that may well cover a lot of things.

**Reply by arron122** [reply]
Thanks for the reply 😄. I haven't but I will and see where it leads me. Gonna put all my brain power and figure it out. Hope you don't me asking questions when they arise 🧐

**Reply by arron122** [reply]
> Hi @arron122
> 
> Have you looked at the API documentation? https://github.com/erew123/alltalk_tts/wiki#-api-documentation that may well cover a lot of things.

Just after your reply I used the javascript code for streaming. Was working fine. Then all of a sudden it stopped working a day or two after. Was there an update or something? Few days now been racking my head trying to figure out what went wrong 🫠

**Reply by erew123** [reply]
Hi @arron122 I am away travelling and unable to test code changes/updates currently. I did make a loose statement here on my Edit 2 about this and the Coqui situation https://github.com/erew123/alltalk_tts/discussions/245#discussioncomment-10838624

The Coqui TTS engine has been updated bu idiap/eginhard https://github.com/idiap/coqui-ai-TTS/releases/tag/v0.24.2

The transfromers version reqirements for Coqui TTS do list as allowing later/newer version of transformers now. I **dont** know if newer versions are required for streaming to operate......

As there is a comment on that release of `XTTS streaming fixes by @eginhard in https://github.com/idiap/coqui-ai-TTS/pull/46 and by @gravityrail in https://github.com/idiap/coqui-ai-TTS/pull/59` but I have no current way to test. 

You are welcome to try figure out the transfomer versions of course, **OR** `start_environment` and then you **should** be able to `pip install coqui-tts==0.24.1` to downgrade to the previous version, though do a `pip show coqui-tts` after to confirm its updated/changed.

I suspect it will be the transformer version requirements, rather than a code change required... but I cant say for now and I cannot test.  I am away, caring for a family member, no access to computer capable of running code/testing. I will of course look more in future, but the above should get you in the correct direction.

**Reply by arron122** [reply]
I kept learning and got it figured out 😄.  Everything works fine. Would have dropped a video clip of the progress but the clip size exceeded the 10MB.
